trieste/acquisition/rule.py

```diff
@@ -944,6 +988,9 @@ def acquire(
 class UpdatableTrustRegion(SearchSpace):
     """A search space that can be updated."""
 
+    def __init__(self) -> None:
+        self.index: Optional[int] = None
```

Comment:
explain what this index is?

Reply:
I have added an explanation (and a new arg), plus renamed it.